### PR TITLE
Upgrade Flax NNX Model Surgery

### PR DESCRIPTION
**Pull request by 8bitmp3**
@IvyZX

### CODE REVIEW

File: docs/nnx/.ipynb_checkpoints/surgery-checkpoint.ipynb
Comment (8bitmp3) [inline docs/nnx/.ipynb_checkpoints/surgery-checkpoint.ipynb:1]: @IvyZX What shall we do about this Surgery Checkpoint file? Did it get created automatically? It doesn't contain the diffs.

File: docs/nnx/surgery.md [hunk 1]
```diff
@@ -42,19 +42,19 @@ class TwoLayerMLP(nnx.Module):
...
 ```
 
-## Pythonic module manipulations
+## Pythonic module manipulation
 
-Model surgery is easiest when you already have a fully fleshed-out model loaded with correct parameters, and you don't intend to change your model definition code.
+Doing model surgery is easiest when you already have a fully fleshed-out model loaded with correct parameters, and you don't intend to change your model definition code.
 
-You can make a variety of pythonic operations on its submodules, like swapping in/out, sharing modules/weights, monkeypatching, etc. See a few code examples below.
+You can perform a variety of Pythonic operations on its sub-modules, such as swapping in/out, sharing modules/weights, monkey-patching, and so on. Below are examples of sub-module sharing, module sharing, variable sharing, and monkey-patching:
```
Comment (IvyZX) [inline docs/nnx/surgery.md:49]: > sub-module sharing, module sharing, variable sharing, and monkey-patching
Repeating this twice seems verbose. Can we do this instead:

> You can perform a variety of Pythonic operations on its sub-modules, such as sub-module swapping, module sharing, variable sharing, and monkey-patching:

File: docs/nnx/.ipynb_checkpoints/surgery-checkpoint.ipynb
Comment (IvyZX) [reply docs/nnx/.ipynb_checkpoints/surgery-checkpoint.ipynb:1]: This seems to be an auto-generated local copy. Can you remove it from your commit?

File: docs/nnx/surgery.md [hunk 2]
```diff
@@ -65,30 +65,30 @@ model.linear2 = model.linear1
...
+# Monkey-patching
 model = TwoLayerMLP(4, rngs=nnx.Rngs(0))
 def awesome_layer(x): return x
 model.linear2 = awesome_layer
 np.testing.assert_allclose(model(x), model.linear1(x))
 
 ```
 
-## Create model and state without memory allocation
+## Creating and manipulating an abstract model or state without memory allocation
```
Comment (IvyZX) [inline docs/nnx/surgery.md:83]: This seems too verbose for a title. Can we drop "and manipulating"?

File: docs/nnx/surgery.md [hunk 3]
```diff
@@ -131,24 +131,24 @@ abs_state['linear1']['bias'].value = model.linear1.bias
 abs_state['linear2']['kernel'].value = model.linear2.kernel
 abs_state['linear2']['bias'].value = model.linear2.bias
 nnx.update(abs_model, abs_state)
-np.testing.assert_allclose(abs_model(x), model(x))  # they are equivalent now!
+np.testing.assert_allclose(abs_model(x), model(x))  # They are equivalent now!
 ```
 
-## Checkpoint surgery
+## Checkpoint surgery: Manipulating parameter states that are incompatible with existing model code
```
Comment (IvyZX) [inline docs/nnx/surgery.md:137]: Same here - a bit too verbose for a title. Checkpoint surgery is a phrase that people use often, so it shouldn't require too much explanation. Maybe we keep the original title?

**Review by IvyZX — COMMENTED** [review]
Thanks for looking into this! Just some small nits.

File: docs/nnx/surgery.md (hunk 2 above)
Comment (8bitmp3) [reply docs/nnx/surgery.md:83]: SGTM!

File: docs/nnx/surgery.md (hunk 3 above)
Comment (8bitmp3) [reply docs/nnx/surgery.md:137]: SGTM!

File: docs/nnx/.ipynb_checkpoints/surgery-checkpoint.ipynb
Comment (8bitmp3) [reply docs/nnx/.ipynb_checkpoints/surgery-checkpoint.ipynb:1]: Done!

File: docs/nnx/surgery.md (hunk 1 above)
Comment (8bitmp3) [reply docs/nnx/surgery.md:49]: SGTM!

**Review by IvyZX — APPROVED** [review]

**Pull request merged**